### PR TITLE
Refactoring \yii\db\Connection::getDriverName

### PR DESCRIPTION
**Pull request by githubjeka**
To bit increase performance.
Old code:

```
$dnsString = 'sqlsrv:Server=localhost;Database=test';

if (($pos = strpos($dnsString, ':')) !== false) {
    echo strtolower(substr($dnsString, 0, $pos));
}
```

Now

```
if (($foundDSN = strstr($dnsString, ':', true)) !== false) {
    echo strtolower($foundDSN);
}
```

|  | Old | Now |
| --- | --- | --- |
| time | 2.1934509277344E-5 | 1.5020370483398E-5 |
| memory | 370328 | 370320 |

|
|

### CODE REVIEW

File: framework/db/Connection.php [hunk 1]
```diff
@@ -811,13 +811,16 @@ function ($matches) {
...
+        if ($this->_driverName !== null) {
+            return $this->_driverName;
         }
+
+        $foundDSN = strstr($this->dsn, ':', true);
+        if ($foundDSN !== false) {
+            $foundDSN = $this->getSlavePdo()->getAttribute(PDO::ATTR_DRIVER_NAME);
+        }
+
+        $this->_driverName = strtolower($foundDSN);
```
Comment (githubjeka) [inline framework/db/Connection.php]: I'm interested to delete this line. 

I have connection error comes from PDO: "the driver can not be found", if it is in uppercase. 

:confused:

**Comment by SilverFire** [comment]
Is it a benchmark per-request?

**Comment by githubjeka** [comment]
No, it's one run by the code. It's in order to make sure that the **strstr **faster.

File: framework/db/Connection.php [hunk 2]
```diff
@@ -811,13 +811,16 @@ function ($matches) {
...
-            if (($pos = strpos($this->dsn, ':')) !== false) {
-                $this->_driverName = strtolower(substr($this->dsn, 0, $pos));
-            } else {
-                $this->_driverName = strtolower($this->getSlavePdo()->getAttribute(PDO::ATTR_DRIVER_NAME));
-            }
+        if ($this->_driverName !== null) {
+            return $this->_driverName;
         }
+
+        $foundDSN = strstr($this->dsn, ':', true);
```
Comment (SilverFire) [inline framework/db/Connection.php]: I'd rename it to `$driverName` to be consistend

**Comment by githubjeka** [comment]
Whay you think about https://github.com/yiisoft/yii2/pull/10818#discussion-diff-52665460 ?

File: framework/db/Connection.php (hunk 1 above)
Comment (SilverFire) [inline framework/db/Connection.php]: I'm OK with thay. @yiisoft/core-developers ?

**Comment by cebe** [comment]
sorry, but I do not see the benefit of this change, this method is called once per request so it does not change anything while code becomes less readable.

**Pull request closed without merging**